cli/internal/prune/prune.go

```diff
@@ -191,7 +193,34 @@ func (p *prune) prune(opts *turbostate.PrunePayload) error {
 	}
 
 	if fs.FileExists("turbo.json") {
-		if err := fs.CopyFile(&fs.LstatCachedFile{Path: p.base.RepoRoot.UntypedJoin("turbo.json")}, fullDir.UntypedJoin("turbo.json").ToStringDuringMigration()); err != nil {
+		turboJSON, err := fs.ReadTurboConfig(".", rootPackageJSON)
+		if err != nil {
+			return errors.Wrap(err, "failed to load turbo.json")
+		}
+
+		// when executing a prune, it is not enough to simply copy the file, as
+		// tasks may refer to scopes that no longer exist. to remedy this, we need
+		// to remove from the Pipeline the TaskDefinitions that no longer apply
+		for key := range turboJSON.Pipeline {
+			shouldRemove := true
+			for _, scope := range opts.Scope {
+				if util.IsTaskInPackage(key, scope) {
+					shouldRemove = false
+					break
+				}
+			}
+			if shouldRemove {
+				delete(turboJSON.Pipeline, key)
+			}
+		}
+
+		bytes, err := turboJSON.MarshalJSON()
+
+		if err != nil {
+			return errors.Wrap(err, "failed to write turbo.json")
+		}
+
+		if err := os.WriteFile(fullDir.UntypedJoin("turbo.json").ToStringDuringMigration(), bytes, 0644); err != nil {
```

Comment:
Coding in GitHub comments, but I think this works:
```suggestion
		if err := fullDir.UntypedJoin("turbo.json").WriteFile(bytes, 0644); err != nil {
```

Reply:
It does indeed, thank you!